doc/decisions/library_directory_structure.md

```diff
@@ -0,0 +1,22 @@
+# Library Directory Structure
+
+## Problem
+
+Currently src/libs/elektra contains the source of several libraries. Each of the libraries should have its own folder.
+Furthermore, we need to allow rust source to exist in parallel.
```

Comment:
I would suggest this structure for the repo (formatted as YAML for highlighting):

```yaml
doc:
examples:
scripts:
src:
 c:
  include: = /src/include (only headers that simply combine other headers)
  libs: = /src/libs (one folder per lib, contains headers and source, only libs written in C)
  plugins: = /src/plugins (one folder per plugin, only plugins written in C)
  tools: = /src/tools (one folder per tool, only tools written in C)
  tests: = /tests (only tests written in C)
 rust:
  binding: = /src/bindings/rust
  libs:
   core: = new Rust alternative of libelektra-core
  _: {other folders depend on Rust's needs, AFAIK tests are normally in the same file as source}
 cpp:
  binding: = /src/bindings/cpp (without the tests)
  libs: (one folder per lib, contains headers and source, only libs written in C++)
  plugins: (one folder per plugin, only plugins written in C++)
  tools: (one folder per tool, only tools written in C++)
  tests: (only tests written in C++)
 _: {folders for other languages, "binding" is always the folder for the binding to the C API, rest depends on the language's needs}
```

That way we can also confine the CMake checks for every language (compiler installed, etc.) to `src/<lang>/CMakeLists.txt` folder.

PS. IMO `benchmarks` should be moved into a separate repo. They are not really relevant to users of Elektra and there is no real technical reason to keep them in this repo. In fact, since benchmark results will only be relevant for a specific version of Elektra, we could use one branch per benchmark and use git submodule pointing to the relevant commit of `libelektra`.